snapshots/rawblock/README.md

```diff
@@ -0,0 +1,34 @@
+## Rawblock snapshotter
+
+Rawblock is a `containerd` snapshotter plugin that stores snapshots as block files on the local file system. It makes
```

Comment:
What are the advantages and disadvantages compared to devmapper?

Reply:
To compare them is to compare dm-thin vs. file system snapshots. Here are what I can find/think of so far:
Pros:
1. Simplicity. dm-thin requires an additional block(btree) mapping from the thin device to the blocks in the data file, and needs a userspace daemon to monitor and increase the size of the thin-pool data file automatically. Rawblock does not have such complexity, -- the snapshot functionality is handled by host file system together with file block mapping. Free disk space is also managed by host file system internally.
2. dm-thin data file did not return disk spaces back to host file system after deleting thin provisionings/snapshots. There is a long standing moby issue for it moby/moby#3182. In theory it can be fixed by fallocate(2) `FALLOC_FL_PUNCH_HOLE` so I'm not sure if it is still the case.

Cons:
1. If host file system does not support reflink, rawblock snapshot creation falls back to plain data copy and can be very slow.
2. dm-thin allows to put metadata and data files separately. So it is possible to put metadata file on a fast storage (e.g., nvme) to gain better metadata performance.
3. dm-thin pool metadata and data files can be raw block devices and thus removing the extra layer of host file system, making it possible to be faster than rawblock snapshotter due to one less file system layer.